src/adaptation/Adaptation.jl

```diff
@@ -7,6 +7,7 @@ import LinearAlgebra, Statistics
 using ..AdvancedHMC: DEBUG
 
 abstract type AbstractAdaptor end
+finalize!(::AbstractAdaptor) = nothing
```

Comment:
Consider replacing this line with a generic function, e.g.

```julia
function finalize!(::AbstractAdaptor)  end
```

Reply:
I'm following this: https://github.com/jrevels/YASGuide
> For single-line method definitions, use the form f(args...) = ....

> If a function does not have a clearly appropriate return value, then explicitly return nothing.

The first one is consistent to ours:

> Only use short-form function definitions when they fit on a single line:

Reply:
Ah, I mean can we try to define a generic function (interface) for abstract types, instead of just return `nothing`? That is,

```julia
finalize!(::AbstractAdaptor) = nothing
```

===>

```julia
function finalize!(::AbstractAdaptor)  end 
```

Reply:
I don't think there is a generic one for `AbstractAdaptor`. E.g. for preconditioner there is not `finalize!` needed but for step size adaptor there is. And thus for composite adaptor we should do `finalize!` to the step size component.

Reply:
Since `function finalize!(::AbstractAdaptor)  end` is a valid Julia function, so we can use it as a "generic" function. For more details, you can take a look at https://pkg.julialang.org/docs/julia/THl1k/1.1.0/manual/methods.html#Empty-generic-functions-1, although it's not entirely the same.

Reply:
I'm confused. So you mean it's more about a styling suggestion?

Reply:
I understand that we can do `function finalize!(::AbstractAdaptor)  end` as well but I kind of prefer the idea that 

> If a function does not have a clearly appropriate return value, then explicitly return nothing.

Reply:
I just tried this myself, tt turns out that 

```julia
function finalize!(::AbstractAdaptor) end
```

produces the same output (ie. `nothing`) as your code

finalize!(::AbstractAdaptor) = nothing

Sorry for the confusion.